pkg/api/validation/validation.go

```diff
@@ -731,6 +731,14 @@ func ValidateService(service *api.Service) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldNotSupported("spec.sessionAffinity", service.Spec.SessionAffinity))
 	}
 
+	for _, ip := range service.Spec.PublicIPs {
+		if !util.IsValidIP(ip) || ip == "0.0.0.0" {
+			allErrs = append(allErrs, errs.NewFieldInvalid("spec.publicIPs", ip, "is not an IP address"))
+		} else if ip == "127.0.0.1" {
```

Comment:
That simplifies logic a lot! Thanks!

Regarding IsGlobalUnicast() - isn't it only for IPv6 which we don't support anyway? IsValidIP returns true only for IPv4 addresses.